### PR TITLE
[No QA][Cleanup] Always use constants for screen names

### PR DESCRIPTION
**Pull request by blazejkustra**
### Details

This PR cleans the usage of screen names throughout the App. Always use constants for screen names in E/App (from SCREENS.ts).

### Fixed Issues
$ https://github.com/Expensify/App/issues/32141
PROPOSAL: N/A

### Tests
- [x] Verify that all pages are linked correctly (looking at the code and testing manually)
- [x] Test the app generally (report screen, login screen, settings, workspaces, profile etc)

### Offline tests
N/A

### QA Steps
N/A

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

[anroid.webm](https://github.com/Expensify/App/assets/46095609/115567b6-5114-484a-9037-67fee3d1abbc)

</details>

<details>
<summary>Android: mWeb Chrome</summary>

[a-web.webm](https://github.com/Expensify/App/assets/46095609/d8a721b1-157d-433a-a577-6eb2730c33bc)

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/46095609/503e71ab-675a-401a-a602-162ba44ec9fa

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/46095609/ef700aab-673e-4cd8-8ed6-3e9bd39fca69

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/46095609/778c2b68-d4c6-4884-bbbc-f2726b0d4b80

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/46095609/2a1c3d96-7664-44cd-916e-993293915181

</details>

### CODE REVIEW

File: contributingGuides/NAVIGATION.md
Comment (kowczarz) [inline contributingGuides/NAVIGATION.md:68]: In a followup PR after the merge we should consider directing to `Expensify/App` repo again.

**Review by kowczarz — APPROVED** [review]
LGTM

File: contributingGuides/NAVIGATION.md
Comment (cubuspl42) [reply contributingGuides/NAVIGATION.md:68]: It can be done in this PR as the last think, can't it? Using the top commit SHA1

Comment (cubuspl42) [reply contributingGuides/NAVIGATION.md:68]: Okey, maybe not the _top_ one, as that's a cycle =P Just the last commit modifying `linkingConfig.ts`

**Review by jakub-trzebiatowski — APPROVED** [review]
I carefully checked the whole diff, searching for manual mistakes, but found none. Looks good.

**Comment by blazejkustra** [comment]
ON HOLD: let’s wait for AppNavigator [PR](https://github.com/Expensify/App/pull/29157) to be merged first

File: src/SCREENS.ts
Comment (roryabraham) [inline src/SCREENS.ts:30]: Can we make `PERSONAL_DETAILS` an object so that the hierarchy is clear? Just like we do for `SETTINGS`?

File: src/SCREENS.ts
Comment (roryabraham) [inline src/SCREENS.ts:58]: Same for PREFERENCES

**Review by roryabraham — COMMENTED** [review]
LGTM. I'm also wondering if we can leverage TS to ensure that this pattern is followed, such that if you try to add a screen name without putting it in SCREENS.ts, or try to reference it elsewhere in the navigation stack without using the constant, then tsc will complain.

**Comment by blazejkustra** [comment]
@cubuspl42 Working on it!

**Review by jakub-trzebiatowski — APPROVED** [review]

**Review by roryabraham — APPROVED** [review]

**Pull request merged**
